apps/icp_app/icp_txn.c

```diff
@@ -404,21 +408,51 @@ static bool sign_txn(der_sig_t *der_signature) {
                           seed,
                           &hdnode);
 
-  uint8_t signature[64];
-  ecdsa_sign_digest(
-      &secp256k1, hdnode.private_key, digest, signature, NULL, NULL);
-
-  der_signature->size = ecdsa_sig_to_der(signature, der_signature->bytes);
-
+  ecdsa_sign_digest(&secp256k1,
+                    hdnode.private_key,
+                    digest,
+                    signature->transfer_req_signature,
+                    NULL,
+                    NULL);
+
+  uint8_t read_state_request_id[SHA256_DIGEST_LENGTH];
```

Comment:
No uninitialized variables
```suggestion
  uint8_t read_state_request_id[SHA256_DIGEST_LENGTH] = {0};
```